File: example-base-28/nodejs-utilities-playground-v2/src/dsa/arrays/merge-sorted-arrays.js
function mergeSortedArrays(first, second) {
  const result = [];
  let i = 0, j = 0;
  while (i < first.length && j < second.length) {
    if (first[i] <= second[j]) result.push(first[i++]);
    else result.push(second[j++]);
  }
  while (i < first.length) result.push(first[i++]);
  while (j < second.length) result.push(second[j++]);
  return result;
}
module.exports = { mergeSortedArrays };

File: example-base-27/java-interview-utilities-complete/src/main/java/com/prem/anagram/MergeSortedArrays.java
package com.prem.anagram;
public class MergeSortedArrays {
    public static int[] merge(int[] first,int[] second) {
        if(first==null) return second==null?new int[0]:second.clone();
        if(second==null) return first.clone();
        int[] result=new int[first.length+second.length];
        int i=0,j=0,k=0;
        while(i<first.length && j<second.length)
            result[k++]=first[i]<=second[j]?first[i++]:second[j++];
        while(i<first.length)result[k++]=first[i++];
        while(j<second.length)result[k++]=second[j++];
        return result;
    }
}
